### PR TITLE
Use `CMAwM` class even when there is no discrete params

### PR DESCRIPTION
**Pull request by c-bata**
## Motivation

Use `CMAwM` even when all elements of steps array are less than or equal with 0 because it makes `_restore_optimizer()` method complicated.

## Description of the changes

In https://github.com/CyberAgentAILab/cmaes/pull/142, `CMAwM` class accepts no discrete search spaces that will be released at v0.9.1.

### CODE REVIEW

**Comment by c-bata** [comment]
I've just made this PR ready for review since cmaes v0.9.1 has been released. Let me assign @knshnb, an author of #4016 

@HideakiImamura Could you assign someone as a second reviewer?

File: tests/samplers_tests/test_cmaes.py [hunk 1]
```diff
@@ -637,13 +637,9 @@ def objective_continuous(trial: optuna.Trial) -> float:
 
     objectives = [objective_discrete, objective_mixed, objective_continuous]
     # When all the seach spaces are continuous, `CMA` is used.
```
Comment (knshnb) [inline tests/samplers_tests/test_cmaes.py:639]: This comment is also out-dated.
```suggestion
```

**Review by knshnb — APPROVED** [review]
LGTM with one small comment!

**Comment by c-bata** [comment]
@HideakiImamura Could you review this PR?

**Review by HideakiImamura — APPROVED** [review]
LGTM.

**Pull request merged**